index.js

```diff
@@ -262,12 +262,28 @@ for (let i = 0; i < buttons.length; i += 1) {
 const formContact = document.querySelector('.footer-form');
 const emailval = document.querySelector('#iemail');
 const errormessage = document.querySelector('.errormessage');
-
+const fname = document.querySelector('#fname');
+const messages = document.querySelector('#isuggest');
 formContact.addEventListener('submit', (event) => {
   const character = /[A-Z]/;
   const emailvalidate = emailval.value;
   if (character.test(emailvalidate)) {
     errormessage.innerHTML = 'Your email shouln\'t contain block letters!';
     event.preventDefault();
   }
-});
\ No newline at end of file
+});
+// preserve contact form
+formContact.addEventListener('input', () => {
+  const storeData = {
+    Name: fname.value,
+    Email: emailval.value,
+    Message: messages.value,
+  };
+  localStorage.setItem('refillForm', JSON.stringify(storeData));
+});
+
+const getData = JSON.parse(localStorage.getItem('refillForm'));
+fname.value = getData.Name;
+emailval.value = getData.Email;
+messages.value = getData.Message;
```

Comment:
- [ ] Good job implementing the features to preserve data :clap:  however, please notice you are not checking for null value initially that's why there is an error in the console :point_down: 
![image](https://user-images.githubusercontent.com/97521557/216588601-8b40169e-bc83-456b-a234-502fe8fa7764.png)
please fix this by implementing if condition like below  :point_down: 
```
if(getData){
fname.value = getData.Name;
emailval.value = getData.Email;
messages.value = getData.Message;
}
```